src/Protocol/ECommerce.php

```diff
@@ -252,8 +264,8 @@ protected function generateSignature(array $data, $encoding = 'UTF-8')
             );
 
             $data['receipt_no'] = ProtocolHelper::mbStrPad($data['receipt_no'], 6, "0", STR_PAD_LEFT, $encoding);
+            $data['msgdata'] = ProtocolHelper::mbStrPad($data['msgdata'], 40, " ", STR_PAD_RIGHT, $encoding, $data['respcode'] === self::PAYMENT_RESPONSE_ABORT);
```

Comment:
We could do:

````php
$data['msgdata'] = ProtocolHelper::mbStrPad(
    $data['respcode'] === self::PAYMENT_RESPONSE_ABORT ? " " : $data['msgdata'], 
    40, 
    " ", 
    STR_PAD_RIGHT, 
    $encoding
);
````
and leave `ProtocolHelper::mbStrPad` as it is. Just wondering if we can avoid `ProtocolHelper::mbStrPad` single responsibility principle violation and not make it more complicated?